### DISCUSSION TITLE
Regarding contact resistance parameter.

### DISCUSSION THREAD

**Discussion by manjunathnilugal** [pybamm-team/PyBaMM#2116] [Q&A] [answered]
Dear All,
I need to implement contact resistance parameter for the electrodes  in the pybamm model.
How to implement this parameter in simulation is there any sub-model available ?
Regards,

**Comment by valentinsulzer** [comment, marked as answer]
Do you have a reference for the (sub)model? Is it just an additional Ohmic resistance?

**Reply by manjunathnilugal** [reply]
Dear Sir, 
I need to include the contact resistance as given below in the source code, Kindly help me where I need to change the equation in the source code of pybamm. I am not able to figure it out where to change the source code. Kindly help me to resolve this issue.
It is an Ohmic resistance which includes resistance at current collector, electrodes contact resistance, and electrolyte resistance.
Anode := 30 ohm.cm2
Cathode := 80 ohm.cm2

<img width="376" alt="image" src="https://user-images.githubusercontent.com/82285213/175368105-035b0d73-5517-4f4f-8ee9-7b61c1ede001.png">

If there are any similar documents, kindly let me know.

**Reply by brosaplanella** [reply]
If it is just an Ohmic resistance it might be easier to postprocess the voltage by doing: `V_IR = solution["Terminal voltage [V]"].entries - solution["Current [A]"].entries * IR`.

**Reply by manjunathnilugal** [reply]
Yes sir, This solution works, but the cut-off voltage gets reduced from 2.5 V to 2.3V if I post process, i,.e pybamm gives me terminal voltage  as 2.5V at the end of the simulation and if I post process it with contact resistance the final voltage reduces to 2.3V which is below the cut-off value. This is causing issue in the drive cycle due to fluctuating current values. So I  wanted it to process this value along simulation. 
I tried increasing the default cut-off  voltage value and then post process, but I am not getting the accurate results during drive cycle.
So, I wanted to process  ` I*R ` in the P2D model itself, which will include the fluctuating current values.
If there are any similar document, kindly share it with me.

**Reply by manjunathnilugal** [reply]
Sir, I tried changing the equation in the Basic_DFN model in the installed pybamm code in the local machine.
V = Terminal_voltage - (Current*Resistance)
path := C:\Users\.....\site-packages\pybamm\models\full_battery_models\lithium_ion\basic_dfn.py

Even after editing the basic_dfn.py code there is no difference in the output. Kindly let me know, if there is anything missing.
**
![image](https://user-images.githubusercontent.com/82285213/180410145-4a45e1b1-f3f8-4222-805e-13cc9c3158fe.png)
**

**Reply by brosaplanella** [reply]
It could be because due to the installation, PyBaMM is still using the original model. You could just use the standard DFN model and after solving postprocess the solution by doing something like `V_IR = solution["Terminal voltage [V]"].entries - solution["Current [A]"].entries * IR`. This will be a numpy array so you can plot it or store it as you wish.

**Reply by brosaplanella** [reply]
Alternatively, you can try to do something along the lines of:
```python3
model = pybamm.lithium_ion.DFN()
V_old = model.variables["Terminal voltage [V]"]
I = model.variables["Current [A]"]
model.variables["Terminal voltage [V]"] = V_old - I * R
```

That should give you the behaviour you want.

**Reply by manjunathnilugal** [reply]
@brosaplanella Thanks a lot sir, This solution helped a lot.

**Comment by manjunathnilugal** [comment]
Sir, In case if I post process the voltage values, we wont be getting the exact lithium concentration in the electrodes.

**Comment by shenxin0307** [comment]
Hi, I noticed that contact resistance had been added after version 23.1. How can I pass this parameter to the DFN model? The below code seems doesn't work.
```
parameter_values = pybamm.ParameterValues("Marquis2019")
parameter_values["Contact resistance [Ohm]"]=1
options = {"thermal":"lumped"}
model = pybamm.lithium_ion.DFN(options)
```

**Reply by brosaplanella** [reply]
This should work. The example above is not complete, ensure that you pass the updated parameter values when running the simulation.

**Reply by shenxin0307** [reply]
I added an option to the model. It works now.
`options = {"thermal":"lumped","contact resistance":"true"}`
